nsv13/code/modules/overmap/fighters/fighters.dm

```diff
@@ -0,0 +1,355 @@
+#define MS_CLOSED 		0
+#define MS_UNSECURE 	1
+#define MS_OPEN	 		2
+
+//Fighter
+
+/obj/structure/overmap/fighter
+	name = "Fighter"
+	desc = "A space faring fighter craft."
+	icon = 'icons/obj/janitor.dmi'
+	icon_state = "mop"
+	brakes = TRUE
+	armor = list("melee" = 80, "bullet" = 80, "laser" = 80, "energy" = 50, "bomb" = 50, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 80) //temp to stop easy destruction from small arms
+	var/maint_state = MS_CLOSED
+	var/piloted = FALSE
+	var/a_eff = 0
+	var/f_eff = 0
+
+/obj/structure/overmap/fighter/prebuilt
+	var/list/components = list(/obj/item/twohanded/required/fighter_component/empennage,
+								/obj/item/twohanded/required/fighter_component/wing,
+								/obj/item/twohanded/required/fighter_component/wing,
+								/obj/item/twohanded/required/fighter_component/landing_gear,
+								/obj/item/twohanded/required/fighter_component/cockpit,
+								/obj/item/twohanded/required/fighter_component/armour_plating,
+								/obj/item/twohanded/required/fighter_component/fuel_tank,
+								/obj/item/fighter_component/avionics,
+								/obj/item/fighter_component/fuel_lines,
+								/obj/item/fighter_component/targeting_sensor,
+								/obj/item/twohanded/required/fighter_component/engine,
+								/obj/item/twohanded/required/fighter_component/engine,
+								/obj/structure/munition/fast,
+								/obj/structure/munition/fast,
+								/obj/item/twohanded/required/fighter_component/primary_cannon)
+
+/obj/structure/overmap/fighter/Initialize()
+	.=..()
+	update_stats()
+	obj_integrity = max_integrity
+
+/obj/structure/overmap/fighter/prebuilt/Initialize()
+	.()
+	name = new_station_name() //temp - replace this with a fighter name list
+	for(var/item in components)
+		new item(src)
+
+/obj/structure/overmap/fighter/proc/update_stats() //PLACEHOLDER JANK SYSTEM
+	var/obj/item/twohanded/required/fighter_component/armour_plating/sap = get_part(/obj/item/twohanded/required/fighter_component/armour_plating)
+	var/obj/item/twohanded/required/fighter_component/fuel_tank/sft = get_part(/obj/item/twohanded/required/fighter_component/fuel_tank)
+	var/obj/item/fighter_component/targeting_sensor/sts = get_part(/obj/item/fighter_component/targeting_sensor)
+	var/obj/item/fighter_component/fuel_lines/sfl = get_part(/obj/item/fighter_component/fuel_lines)
+	var/senc = 0
+	var/sens = 0
+	var/sene = 0
+	for(var/obj/item/twohanded/required/fighter_component/engine/sen in contents)
+		senc++
+		sens = sens + sen.speed
+		sene = sene + sen.consumption
+	sens = sens / senc
+	sene = sene / senc
+	f_eff = sene + sfl.fuel_efficiency / 2
+	a_eff = sts.weapon_efficiency
+	max_integrity = 100 * sap.armour
+	create_reagents(sft.capacity) //DANGER - TEMP
+
+/obj/structure/overmap/fighter/process()

```

Comment:
Do this on "slowprocess", it's an overmap specific function which only processes at a slower rate, rather than multiple times per second

Reply:
Slowprocess for where I'm going to add the internal cockpit stuffs?

Reply:
We don't want to process atmos that quickly or performance may suffer

Reply:
Its how mecha do it currently, but sure I can write it in there.